src/internal/m365/onedrive/restore.go

```diff
@@ -1215,51 +1221,63 @@ func ensureDriveExists(
 	pdagrf PostDriveAndGetRootFolderer,
 	caches *restoreCaches,
 	drivePath *path.DrivePath,
-	protectedResourceID, driveName string,
-) error {
+	protectedResourceID, fallbackDriveName string,
+) (driveInfo, error) {
 	driveID := drivePath.DriveID
 
-	// the drive might already be cached
-	if _, ok := caches.DriveIDToDriveInfo[driveID]; ok {
-		return nil
+	// the drive might already be cached by ID.  it's okay
+	// if the name has changed.  the ID is a better reference
+	// anyway.
+	if di, ok := caches.DriveIDToDriveInfo[driveID]; ok {
+		return di, nil
 	}
 
 	var (
-		newDriveName = driveName
+		newDriveName = fallbackDriveName
 		newDrive     models.Driveable
 		err          error
 	)
 
-	if _, ok := caches.DriveNameToDriveInfo[newDriveName]; ok {
-		newDriveName = fmt.Sprintf("%s %d", driveName, 1)
+	// if the drive wasn't found by ID, maybe we can find a
+	// drive with the same name but different ID.
+	// start by looking up the old drive's name
+	oldName, ok := caches.BackupDriveIDName.NameOf(driveID)
```

Comment:
IIUC `caches.BackupDriveIDName` will only contain the drives that were available at the time of backup, right? Should we get the latest set of drives before we do this check?

Reply:
The latest set of drives are already cached in `caches.DriveIDToDriveInfo`.